lib/init/rc.lib

```diff
@@ -38,8 +38,32 @@ parse_crypttab() {
             break
         }
 
-        # Turn 'UUID=*' lines into device names.
-        [ "${dev##UUID*}" ] || dev=$(blkid -l -o device -t "$dev")
+        # Turn 'UUID=*', 'LABEL=*', 'PARTUUID=*' lines into device names.
+        case ${dev%%=*} in
+            UUID|LABEL|PARTUUID)
+                for line in $(blkid); do
+                    case ${line%%=*} in
+                        /dev/*)
+                            _dev=${line%:}
+                        ;;
+                        UUID|LABEL|PARTUUID)
+                            [ "$_dev" ] && {
```

Comment:
Maybe better to just do: `[ "$_dev" ] || continue` 

Avoids the indentation and braces.

Reply:
Ok, should i add extra space between `[ ]` condition and variables?

Reply:
Yeah